### ISSUE TITLE
Starting daemon on non default port

### ISSUE THREAD

**Issue by dominicpaul1911** [textileio/js-textile-go-daemon#6]
When starting the Daemon on non-default, the subsequent api calls fallback to default ports. Steps I followed are:

`1. let daemon = await new DaemonFactory(binary).spawn({disposable: false, repoPath: repoPath}).catch(err => log.error(err.toString()))`

`2. await daemon.init(account.keypair.secret(), {repoPath: repoPath, swarmPorts: [4102], apiBindAddr: "127.0.0.1:41600", gatewayBinAddr: "127.0.0.1:5150"}).catch(err => log.error('Could not init', err.toString()))`

`3. await daemon.start({repoPath: repoPath}).catch(err => log.error('Not started', err.toString()))`

`4. log.info(await daemon.api.config.get())`

Step 4 displays the api port as the default port and any further api calls go to that port.

**Comment by asutula** [comment]
@carsonfarmer do you have any immediate thoughts about this? If so, great, if not, I'll jump into it.

**Comment by carsonfarmer** [comment]
Not really. I think probably what's going on here is that the js-http-client is hardcoded to use the default ports etc. So it really just needs to get this information from the daemon on startup and use those instead.

**Comment by carsonfarmer** [comment]
@dominicpaul1911 Could you confirm that the init is indeed writing the correct ports to the `textile` config file? Should be a file called `textile` in the repoPath you specified. You can probably share that here, just make sure you don't share any private information 🤓 .

**Comment by dominicpaul1911** [comment]
Attaching textile file 
[textile.txt](https://github.com/textileio/js-textile-go-daemon/files/3893989/textile.txt)

**Comment by carsonfarmer** [comment]
Here's the relevant bit:

```
"Addresses": {
        "API": "\"41600\"",
        "CafeAPI": "0.0.0.0:40601",
        "Gateway": "\"127.0.0.1:5150\"",
        "Profiling": "127.0.0.1:6060"
    },
```

Which is all messed up. Can you share the code snippet used to set these? Maybe just a tiny reproducible example?

**Comment by carsonfarmer** [comment]
The API addr should be something like `"http://127.0.0.1:41600"` and similarly the gateway. It looks like they are getting string escaped, which might be a bug?
